server/fly.io/src/main.rs

```diff
@@ -0,0 +1,79 @@
+use std::collections::BTreeMap;
+use std::net::SocketAddr;
+
+use axum::{
+    extract::Query,
+    routing::{get, post},
+    Router,
+};
+use influxrs::{InfluxClient, Measurement};
+
+fn main() {
+    let rt = tokio::runtime::Runtime::new().unwrap();
+    let task = rt.spawn(async move {
+        let app = Router::new()
+            .route("/", get(root))
+            .route("/record-install", get(record_install))
+            .route("/record-install", post(record_install));
+
+        // ipv6 + ipv6 any addr
+        let addr = SocketAddr::from(([0, 0, 0, 0, 0, 0, 0, 0], 8080));
+        axum::Server::bind(&addr)
+            .serve(app.into_make_service())
+            .await
+            .unwrap();
+    });
+    rt.block_on(task).unwrap();
+}
+
+async fn root() -> &'static str {
+    "Hello, World!"
+}
+
+fn get_env(key: &str) -> String {
+    std::env::var(key).expect(&format!("{key} must be set"))
+}
+
+async fn record_install(Query(params): Query<BTreeMap<String, String>>) -> String {
+    println!("Hi there {params:?}");
+
+    // FIXME: make this in main and pass it down or something?
+    let url = get_env("INFLUX_URL");
+    let token = get_env("INFLUX_TOKEN");
+    let org = get_env("INFLUX_ORG");
+    let bucket = get_env("INFLUX_BUCKET");
+    let client = InfluxClient::builder(url, token, org).build().unwrap();
+
+    let mut point = Measurement::builder("counts").field("count", 1);
+    for (tag, value) in &params {
+        point = point.tag(tag, &**value)
+    }
+    client
+        .write(&bucket, &[point.build().unwrap()])
+        .await
+        .unwrap();
+    format!("Hi there {params:?}")
```

Comment:
I think we can `tokio::spawn` this part, so that the http request would return ASAP (not having to wait for the influx db stuff to be done)

Reply:
I've had trouble recently with things like google cloud run suspending your compute as soon as you return from your http request, so I'm a bit wary of doing this (yes I know the vercel stats server does all sorts of nonsense like this, and fly.io isn't as aggressive as cloud run). I think I prefer structured concurrency these days.